sky/server/requests/payloads.py

```diff
@@ -139,6 +134,19 @@ def to_kwargs(self) -> Dict[str, Any]:
         return kwargs
 
 
+class ValidateBody(DagRequestBody):
+    """The request body for the validate endpoint."""
+    dag: str
+    request_options: Optional[admin_policy.RequestOptions]
+
+
+class OptimizeBody(DagRequestBody):
+    """The request body for the optimize endpoint."""
+    dag: str
+    minimize: common_lib.OptimizeTarget = common_lib.OptimizeTarget.COST
+    request_options: Optional[admin_policy.RequestOptions]
```

Comment:
Can we try if a API server exists, and a user upgrade to the current PR, will the CLIs show clear warning/error that the api server needs to be restarted? We may want to avoid pydantic errors.

Reply:
The case you mentioned works (Old API server, new client), but the reverse (New API server, old client) did not. I've bumped up the API version to 3 to throw a clean error.